services/transactionService.ts

```diff
@@ -1,70 +1,217 @@
-import Web3 from 'web3'
```

Comment:
I changed the order of functions to use https://dzone.com/articles/the-stepdown-rule for increase readability of code